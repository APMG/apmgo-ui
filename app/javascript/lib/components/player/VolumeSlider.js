// @flow
import React from 'react'
import { connect, dispatch } from 'react-redux'
import { volumeChange } from '../../redux/audio-player'

type VolumeSliderProps = {
  volume: number,
  updateVolume: (volume: number) => {}
}

export default function(props: VolumeSliderProps) {

  const updateVolume = (event: any) => {
    props.updateVolume(event.target.value)
  }

  return (
    <input
      type='range'
      value={props.volume}
      onInput={updateVolume}
      onChange={updateVolume}
      min={0}
      max={1}
      step={.01}
    />
  )
}
