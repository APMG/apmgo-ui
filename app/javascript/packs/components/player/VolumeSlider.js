// @flow
import * as React from 'react'
import Slider from 'material-ui/Slider'

type VolumeSliderProps = {
  audio: HTMLAudioElement
}

export default class VolumeSlider extends React.Component {

  props: VolumeSliderProps

  state: {
    volume: number
  }

  constructor(props: VolumeSliderProps) {
    let volume = 1
    if (props.audio && props.audio.volume) {
      volume = props.audio.volume
    }

    super(props)
    this.state = {
      volume: volume
    }
  }

  _updateVolumeInState() {
    this.setState({
      volume: this.props.audio.volume
    })
  }

  _volumeChange(event: any, newValue: number) {
    if (!this.props.audio) {
      return
    }

    this.props.audio.volume = newValue
    this._updateVolumeInState()
  }

  render() {
    return (
      <Slider
        value={ this.state.volume }
        onChange={ this._volumeChange.bind(this)}
      />
    )
  }
}
