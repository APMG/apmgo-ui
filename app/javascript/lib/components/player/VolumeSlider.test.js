import React from 'react'
import VolumeSlider from './VolumeSlider'

import { getSnapshotJson } from '../../__tests__/testHelpers'

describe('VolumeSlider Component', () => {
  describe('Presenter', () => {
    it('Renders', () => {
      let tree = getSnapshotJson(<VolumeSlider/>)
      expect(tree).toMatchSnapshot()
    })
  })
})
