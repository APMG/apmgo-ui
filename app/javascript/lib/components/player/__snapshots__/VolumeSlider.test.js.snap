// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VolumeSlider Component Presenter Renders 1`] = `
<input
  aria-label="Volume"
  max={1}
  min={0}
  onChange={[Function]}
  onInput={[Function]}
  step={0.01}
  type="range"
  value={undefined}
/>
`;
